### PR TITLE
fix: empty loan state

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Properly handles no loan state.

`develop`:

Screenshot 2024-09-24 at 12 33 12

This PR:

Screenshot 2024-09-24 at 12 31 28

This change is needed as we can still have `data` on a position without an actual loan:

Screenshot 2024-09-24 at 12 12 26

## Issue (if applicable)

N/A, discovered by @gomesalexandre in a recent review: https://github.com/shapeshift/web/pull/7753#pullrequestreview-2306156963

## Risk

Low

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

None

## Testing

- When connected to a wallet with no loans, the placeholder component is shown as in the screenshot above.
- When connect to w wallet with loans, the loans are still shown

### Engineering

☝️

### Operations

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

☝️

## Screenshots (if applicable)

See above.

### CODE REVIEW

File: src/pages/Lending/hooks/useAllLendingPositionsData.tsx [hunk 1]
```diff
@@ -115,7 +115,23 @@ export const useAllLendingPositionsData = ({ assetId }: UseAllLendingPositionsDa
...
+      if (!data) return false
+      const {
+        collateralBalanceCryptoPrecision,
+        collateralBalanceFiatUserCurrency,
+        debtBalanceFiatUserCurrency,
+      } = data
+      return (
+        (collateralBalanceCryptoPrecision && parseFloat(collateralBalanceCryptoPrecision) > 0) ||
+        (collateralBalanceFiatUserCurrency && parseFloat(collateralBalanceFiatUserCurrency) > 0) ||
+        (debtBalanceFiatUserCurrency && parseFloat(debtBalanceFiatUserCurrency) > 0)
```
Comment (gomesalexandre) [inline src/pages/Lending/hooks/useAllLendingPositionsData.tsx:131]: Think we're safe using only one of the two first as a criteria if we want to keep things simpler (no need to check for fiat if we check for crypto balance), though just checking for the debt should also be enough - if debt is repaid, there is no loan

**Review by gomesalexandre — APPROVED** [review]
Does precisely what it says on the box!

https://jam.dev/c/ff471b23-f24e-4034-b461-fc50d53019d3

One comment to reduce diff size at your convenience.

**Review by NeOMakinG — APPROVED** [review]
image

Does what it says, lgtm

File: src/pages/Lending/hooks/useAllLendingPositionsData.tsx (hunk 1 above)
Comment (0xApotheosis) [reply src/pages/Lending/hooks/useAllLendingPositionsData.tsx]: Yeah fair enough, simplified in https://github.com/shapeshift/web/pull/7792/commits/4de0861b62fface20bf0cdbf22f415cc7c5c296f 🙏

**Pull request merged**